### PR TITLE
Print target kubeconfig

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
This PR adds the command `target kubeconfig` to print the kubeconfig for the current target.

This is especially useful to retrieve the gardenlogin kubeconfig for `Shoot` clusters, as these kubeconfigs will not be written anymore to the garden cluster as ConfigMaps, as the `gardenlogin-controller-manager` will be deprecated (ref #149)

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
You can now print the kubeconfig for the current target using `gardenctl target kubeconfig` command
```

### CODE REVIEW

File: pkg/cmd/target/kubeconfig.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	if err != nil {
+		return err
+	}
+
+	b, err := writeRawConfig(config)
+	if err != nil {
+		return err
+	}
+
+	_, err = o.IOStreams.Out.Write(b)
```
Comment (holgerkoser) [inline pkg/cmd/target/kubeconfig.go:89]: fmt.Fprintln(o.IOStreams.Out.Write, string(b))

[the remark this answers is not in the document]
Comment (holgerkoser) [reply pkg/cmd/target/kubeconfig.go:89]: or ignore the error. If this no works the user does no see errors
```go
fmt.Fprintf(o.IOStreams.Out, "%s\n", b)

return nil
```

File: pkg/cmd/target/target.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ gardenctl target value/that/matches/pattern --control-plane`,
 
 	cmd.AddCommand(NewCmdUnset(f, NewUnsetOptions(ioStreams)))
 	cmd.AddCommand(NewCmdView(f, NewViewOptions(ioStreams)))
+	cmd.AddCommand(NewCmdKubeconfig(f, NewKubeconfigOptions(ioStreams)))
```
Comment (holgerkoser) [inline pkg/cmd/target/target.go:51]: I think it is better to have this command at top level. It is a command the uses the current target and does something with this information like `ssh` or `k-env` or `p-env`. They are also not subcommands of target.

Comment (holgerkoser) [reply pkg/cmd/target/target.go:51]: Or we create a new top level `get` command (like kubectl) and add this one as first subcommand. In the future we could add a `target` subcommand and deprecate `target view`.

File: docs/help/gardenctl_target_kubeconfig.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+gardenctl target kubeconfig [flags]
+```
+
+### Options
+
+```
+  -h, --help            help for kubeconfig
+  -o, --output string   One of 'yaml' or 'json'.
```
Comment (holgerkoser) [inline docs/help/gardenctl_target_kubeconfig.md:13]: Handle the output flag in run

File: pkg/cmd/target/kubeconfig.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (petersutter) [reply pkg/cmd/target/kubeconfig.go:89]: ✅

File: pkg/cmd/target/target.go (hunk 2 above)
Comment (petersutter) [reply pkg/cmd/target/target.go:51]: ✅

**Pull request closed without merging**